src/main/java/com/learning/mfscreener/mapper/MfSchemeDtoToEntityMapper.java

```diff
@@ -23,13 +23,15 @@
 public abstract class MfSchemeDtoToEntityMapper {
 
     private final Logger log = LoggerFactory.getLogger(this.getClass());
+
     // Define the regular expressions
     private static final Pattern TYPE_CATEGORY_SUBCATEGORY_PATTERN =
```

Comment:
@coderabbitai, how ensure that the regex used here, which is vulnerable to polynomial runtime due to backtracking, cannot lead to denial of service.